frontend/src/maps/MapView.tsx

```diff
@@ -10,20 +10,62 @@ import UserFocusView from "./widgets/UserFocusView.tsx";
 import PanelButton from "./widgets/PanelButton.tsx";
 import MapPanel from "./MapPanel.tsx";
 import { useRouteOperations } from "./manage/operations.ts";
+import { useEffect } from "react";
 
-const MapView = () => {
+import type { Point, RouteSegment } from "./manage/structure";
+import { useRef } from "react";
+
+const MapView = ({
+  onRouteDataChange,
+  initialData,
+}: {
+  onRouteDataChange: (data: {
+    points: Point[];
+    segments: RouteSegment[];
+  }) => void;
+  initialData: { points: Point[]; segments: RouteSegment[] };
+}) => {
   const [panelOpen, setPanelOpen] = useState(false);
+
   const {
     points,
     segments,
     addPoint,
     removePoint,
     reorderPoints,
     togglePointType,
-  } = useRouteOperations();
+  } = useRouteOperations(initialData.points, initialData.segments);
+
+  const hasInteractedRef = useRef(false);
```

Comment:
We should really move hasInteractedRef logic into operations.ts, especially after we remove the need of onRouteDataChange.